### PR TITLE
chore(aws): Add AWSSessionTokenExpired

### PR DESCRIPTION
**Pull request by jfagoagas**
### Description

Add `AWSSessionTokenExpired` exception.

### Checklist

- Are there new checks included in this PR? **No**
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed. -> **No**

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by pedrooot — APPROVED** [review]
🥷 Thanks!

File: prowler/providers/aws/aws_provider.py [hunk 1]
```diff
@@ -1200,6 +1206,12 @@ def test_connection(
                 raise invalid_account_credentials_error
             return Connection(error=invalid_account_credentials_error)
 
+        except AWSSessionTokenExpired as session_token_expired:
+            logger.error(str(session_token_expired))
```
Comment (MrCloudSec) [inline prowler/providers/aws/aws_provider.py:1210]: Why not using our logging pattern?
```suggestion
            logger.error(f"{session_token_expired.__class__.__name__}[{session_token_expired.__traceback__.tb_lineno}]: {session_token_expired})"
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please view my comment.

File: prowler/providers/aws/aws_provider.py (hunk 1 above)
Comment (MrCloudSec) [reply prowler/providers/aws/aws_provider.py:1210]: We will review all our logs in the future.

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**